ffi/src/lib.rs

```diff
@@ -246,6 +378,86 @@ pub struct Extern {
     __marker: core::marker::PhantomData<(*mut u8, core::marker::PhantomPinned)>,
 }
 
+/// Define the correct [`FfiOutput::ReturnType`] out of the given [`COutput::ReturnType`].
+/// The only situation when this is evident is when [`Ir::Type`] is set to [`Transparent`]
+///
+/// Example:
+///
+/// ```
+/// use iroha_ffi::FfiType;
+///
+/// #[derive(FfiType)]
+/// #[ffi_type(unsafe {robust})]
+/// #[repr(transparent)]
+/// struct Example(u32);
+///
+/// /*
+/// Due to the fact that implementations of traits for [`Transparent`] structures delegate to the inner
+/// type `FfiType`, macro expansion will produce the following implementation of [`COutput`] for `Example`:
+///
+/// impl COutput for Example {
+///     type ReturnType = u32;
+/// }
+///
+/// which is corrected via [`ReturnTypeOf`] so that functions generated by [`ffi_import`] return [`Example`]
+/// */
+/// ```
+///
+pub trait ReturnTypeOf<T> {
```

Comment:
I'm quite unhappy that I had to introduce this trait, but on the upside I think all implementations that will ever be needed have been covered in this PR